### DISCUSSION TITLE
Is it possible to manually erase a selection from a combo box?

### DISCUSSION THREAD

**Discussion by louh** [uswds/uswds#4391] [Q&A] [answered]
We're implementing the combo box component and came across something while internally testing its usability versus a third-party autocomplete library that we were previously using.

Given a combo box component that has an "empty" value in its list of options, but a value has been selected, e.g.:

```html
<div class="usa-combo-box">
  <select class="usa-select" name="example" id="example">
    <option value></option>
    <option value="0">not selected</option>
    <option value="1" selected>selected</option>
  </select>
</div>
```

Currently, this would display "selected" in the input box. If a user places their cursor in the input, then manually presses backspace until the input is empty, then presses enter, the component will return to the previous value of "selected", rather than the desired input of empty.

While the user could have achieved the desired result by clicking the X button the appears when an option is selected, we are transitioning from a previous external third-party autocomplete library where clearing the input via keyboard activity was allowed, and users may be confused by the change in behavior.

So my questions are:

- Is there another usability concern that we're overlooking by making it possible to select an empty value by clearing the input with "backspace" or "delete" ?
- Is there a way of achieving what we're looking for with the USWDS combo box component? (I am happy to help work on a solution, if needed.)

Thanks!

**Comment by louh** [comment]
Looking into the HTML generated by the combo box component, I see that an `<option>` with an empty value isn't rendered in the list items.

Another thing I've realized is that if a `<option selected>` is in the original HTML, it won't be the default value in the combo-box unless the `data-default-value` attribute is also set. Is that also done on purpose?

I should note that I'm currently on USWDS 2.9.0, and we're using it vanilla (no JS framework), in a Django app.

**Comment by mejiaj** [comment, marked as answer]
Hey @louh! These are all good points and we welcome any contributions. I've created issues based on these points. 

- Users should be able to hit backspace or delete to erase a selection [[issue #4392]](https://github.com/uswds/uswds/issues/4392)
- Empty option _should_ be rendered [[issue #4393]](https://github.com/uswds/uswds/issues/4393)
- Selected attribute should work like `data-default-value` [[issue #4394]](https://github.com/uswds/uswds/issues/4394)

**Comment by ryparker** [comment]
I retested this against USWDS 3.14.0. Deleting the input text and pressing Enter still restores the existing selection. The clear button clears both the visible input and the underlying select; it is a keyboard-focusable button labeled “Clear the select contents.”

The other two observations also remain relevant: an empty-value option is omitted from the generated list, and initial selection uses `data-default-value` on `.usa-combo-box`, rather than an option's `selected` attribute.

For example:

```html
<div class="usa-combo-box" data-default-value="1">
  <!-- Labeled select and options -->
</div>
```

The issues linked in the earlier answer were closed without implementing those changes. Their closure should not be read as confirmation that an upgrade fixes them. The built-in clear button is the current clearing mechanism; Backspace/Delete alone is still a behavior gap.
